apollo-client/src/main/java/com/ctrip/framework/apollo/spring/annotation/ApolloAnnotationProcessor.java

```diff
@@ -54,7 +57,9 @@ protected void processMethod(final Object bean, String beanName, final Method me
     ReflectionUtils.makeAccessible(method);
     String[] namespaces = annotation.value();
     String[] annotatedInterestedKeys = annotation.interestedKeys();
-    Set<String> interestedKeys = annotatedInterestedKeys.length > 0 ? Sets.newHashSet(annotatedInterestedKeys) : null;
+    String[] annotatedInterestedKeyPatterns = annotation.interestedKeyPatterns();
+    Set<String> interestedKeys = new HashSet<>(Arrays.asList(annotatedInterestedKeys));
+    Set<String> interestedKeyPatterns = new HashSet<>(Arrays.asList(annotatedInterestedKeyPatterns));
```

Comment:
most of the time, the interested keys or patterns are empty, which means we don't need to create an arraylist and a hashset at all time.